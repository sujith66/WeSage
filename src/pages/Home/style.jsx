import styled from "styled-components";
import Button from "@mui/material/Button";

export const HomeContainer = styled("main")(() => ({
  display: "flex",
  '.MuiTypography-h5': {
      color: 'white'
  },
  
  '.transfer__window': {
      backgroundColor: 'white',
      height: '24rem',
      width: '30rem',
      borderRadius: '20px',
      paddingTop: '20px',
      paddingBottom: '20px',
      '@media (max-width:767px)': {
          position: 'absolute',
          width: '15rem'
      }
  },
  '.transfer__main': {
      display: "flex",
      flexDirection: 'column'
  },
    '.MuiButton-contained': {
        borderRadius: '20px'
    },
    '.download__top-icon': {
        fontSize: '12px'
    },
    '.transfer__filelist': {
        display: "flex",
        borderBottom: "1px solid",
        borderTop: "1px solid",
        padding: "4px",
    }
  
}));

export const HeaderContainer = styled("header")(() => ({
  ".MuiPaper-root": {
    backgroundColor: "transparent",
  },
  ".MuiTypography-h6": {
    fontWeight: "bolder",
    fontSize: "30px",
    color: "white",
    fontStyle: "italic",
    marginRight: '30rem'
  },
  nav: {
    background: "white",
    borderRadius: "7px",
'ul,li': {
    display: 'flex'
},
li: {
    borderRight: '1px solid rgba(23,24,26,.11)'
},
    a: {
      textDecoration: "none",
    },
  },
}));

export const ButtonContainer = styled(Button)(() => ({
    marginTop: '2rem'
}));