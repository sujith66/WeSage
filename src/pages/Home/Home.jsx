import * as React from "react";
import CssBaseline from "@mui/material/CssBaseline";
import Typography from "@mui/material/Typography";
import GlobalStyles from "@mui/material/GlobalStyles";
import Container from "@mui/material/Container";
import { HomeContainer, ButtonContainer } from "./style";
import Transfer from "../../components/Images/Transfer/Transfer";
import Download from "../../components/Images/Download/Download";
import Header from "../../components/Header/Header";

function Home() {

  const handleDownload = ()=>{
    console.log('Starting download')
  };

  return (
    <Container>

      {/* Global style component  for handling css resets */}
      <GlobalStyles
        styles={{ ul: { margin: 0, padding: 0, listStyle: "none" } }}
      />
      <CssBaseline />
      <Header />

      {/* Container for handling Home component elements */}
      <HomeContainer
        disableGutters
        maxWidth="sm"
        component="main"
        sx={{ pt: 8, pb: 6 }}
      >
        <div className="transfer__window">
          <Transfer />
          <div className="download__top-icon">
            <h3 className="margin__default">Ready when you are</h3>
            <p className="margin__default">Transfer expires in four days</p>
          </div>
          <p>GoProposal Frontend test </p>

          <div className="transfer__filelist">
            <div>
              <h4 className="margin__default" style={{ fontSize: "14px " }}>
                recruitment-frontend-test.zip
              </h4>
              <p className="margin__default" style={{ display: "flex" }}>
                652kb zip
              </p>
            </div>
            <Download color />
          </div>

          {/* Container for handlng button styles */}
          <ButtonContainer variant="contained" onClick={handleDownload}>Download</ButtonContainer>
        </div>
        <div className="transfer__main">
          <Typography
            component="h1"
            variant="h2"
            align="center"
            color="text.primary"
            gutterBottom
          >
            Transfer your data
          </Typography>
          <Typography
            variant="h5"
            align="center"
            color="text.secondary"
            component="p"
          >
            Quickly build an effective transfer table for your potential
            customers with this layout. It&apos;s built with default MUI
            components with little customization.
          </Typography>
        </div>
      </HomeContainer>
    </Container>
  );
}

export default Home;
