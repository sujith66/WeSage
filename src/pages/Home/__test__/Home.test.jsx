import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Home from "../Home";

const mockHome = ()=>(    <Home />);


//Test suite which will contain all the tests for Home
describe("Testing Home", () => {
  test("check if the  Home component is rendered successfully", () => {
    render(mockHome());
    expect(screen.getByText("Help")).toBeInTheDocument();
  });

  
  test("check if the  tranfer window headers are rendered", () => {
    render(mockHome());
    expect(screen.getByText("Ready when you are")).toBeInTheDocument();
  });

  test('check if the downloa button is rendered',() => {
    render(mockHome());
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("matches the snapshot to check if the component renders correctly", () => {
    const tree = renderer
      .create((mockHome()))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});