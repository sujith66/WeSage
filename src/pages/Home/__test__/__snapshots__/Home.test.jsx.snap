// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home matches the snapshot to check if the component renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <header
      className="sc-jRQBWg jfEDmn"
      elevation={0}
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t018ql-MuiToolbar-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-uh02xs-MuiTypography-root"
        >
          we
        </h6>
        <nav>
          <ul>
            <li>
              <a
                className="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-1xkhg5k-MuiTypography-root-MuiLink-root"
                href="#"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Help
              </a>
            </li>
            <li>
              <a
                className="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-1xkhg5k-MuiTypography-root-MuiLink-root"
                href="#"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Company
              </a>
            </li>
            <li>
              <a
                className="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-1xkhg5k-MuiTypography-root-MuiLink-root"
                href="#"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Pricing
              </a>
               
            </li>
            <li>
              <a
                className="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-1xkhg5k-MuiTypography-root-MuiLink-root"
                href="#"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Sign up
              </a>
               
            </li>
            <li>
              <a
                className="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-1xkhg5k-MuiTypography-root-MuiLink-root"
                href="#"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Log in
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
  </div>
  <main
    className="sc-bdvvtL knzQyN"
  >
    <div
      className="transfer__window"
    >
      <svg
        className="sc-hKwDye bWhgQa"
        viewBox="0 0 170 170"
      >
        <g
          fill="#d4d7d9"
          fillRule="evenodd"
        >
          <path
            d="M145.104 24.896c33.195 33.194 33.195 87.014 0 120.208-33.194 33.195-87.014 33.195-120.208 0C-8.3 111.91-8.3 58.09 24.896 24.896 58.09-8.3 111.91-8.3 145.104 24.896zm-7.071 7.071c-29.29-29.29-76.777-29.29-106.066 0-29.29 29.29-29.29 76.777 0 106.066 29.29 29.29 76.777 29.29 106.066 0 29.29-29.29 29.29-76.777 0-106.066z"
          />
          <path
            d="M82 100.843V59.007A4.006 4.006 0 0 1 86 55c2.21 0 4 1.794 4 4.007v41.777l15.956-15.956a4.003 4.003 0 0 1 5.657 0 4.004 4.004 0 0 1 0 5.657l-22.628 22.628a3.99 3.99 0 0 1-3.017 1.166 3.992 3.992 0 0 1-3.012-1.166L60.328 90.485a4.003 4.003 0 0 1 0-5.657 4.004 4.004 0 0 1 5.657 0L82 100.843z"
          />
        </g>
      </svg>
      <div
        className="download__top-icon"
      >
        <h3
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Ready when you are
        </h3>
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Transfer expires in four days
        </p>
      </div>
      <p>
        GoProposal Frontend test 
      </p>
      <div
        style={
          Object {
            "borderBottom": "1px solid",
            "borderTop": "1px solid",
            "display": "flex",
            "padding": "4px",
          }
        }
      >
        <div>
          <h4
            style={
              Object {
                "fontSize": "14px ",
                "margin": "0",
              }
            }
          >
            recruitment-frontend-test.zip
          </h4>
          <p
            style={
              Object {
                "display": "flex",
                "margin": 0,
              }
            }
            y={true}
          >
            652kb zip
          </p>
        </div>
        <svg
          className="sc-eCImPb jItAHw"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 12c0-5.523-4.477-10-10-10S2 6.477 2 12s4.477 10 10 10 10-4.477 10-10zm-9 1.492V7.998C13 7.446 12.552 7 12 7c-.556 0-1 .447-1 .998v5.48l-2.53-2.53c-.385-.385-1.022-.39-1.413.002-.393.393-.39 1.022-.002 1.412l4.247 4.247c.192.19.447.288.702.288.26.003.514-.095.708-.29l4.247-4.246c.383-.385.387-1.022-.003-1.412-.394-.393-1.023-.392-1.413-.002L13 13.492zM0 12C0 5.373 5.373 0 12 0s12 5.373 12 12-5.373 12-12 12S0 18.627 0 12z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root sc-dkPtRN jhgYgd css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Download
      </button>
    </div>
    <div>
      <h1
        className="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-10759ka-MuiTypography-root"
      >
        Pricing
      </h1>
      <p
        className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1bqvlh8-MuiTypography-root"
      >
        Quickly build an effective pricing table for your potential customers with this layout. It's built with default MUI components with little customization.
      </p>
    </div>
  </main>
</div>
`;
